src/main/scala/cromwell/engine/backend/local/LocalBackend.scala

```diff
@@ -74,4 +103,70 @@ class LocalBackend extends Backend {
       taskOutput.name -> rawValue.map { possiblyAutoConvertedValue(taskOutput, _) }
     }.toMap
   }
+
+  /**
+   * Creates host execution directory, inputs path, and outputs path.  Stages any input files into the workflow-inputs
+   * directory and localizes their paths relative to the container.
+   */
+  override def initializeForWorkflow(descriptor: WorkflowDescriptor): HostInputs = {
+    val hostExecutionDirectory = hostExecutionPath(descriptor).toFile
+    hostExecutionDirectory.mkdirs()
+
+    val hostExecutionAbsolutePath = hostExecutionDirectory.getAbsolutePath
+    Array("workflow-inputs", "workflow-outputs") foreach { Paths.get(hostExecutionAbsolutePath, _).toFile.mkdir() }
+    stageWorkflowInputs(descriptor)
+  }
+
+  /**
+   * {{{cromwell-executions + workflow.name + workflow.id = cromwell-executions/three-step/0f00-ba4}}}
+   */
+  private def hostExecutionPath(workflow: WorkflowDescriptor): Path = Paths.get("cromwell-executions", workflow.name, workflow.id.toString)
+
+  /**
+   * Given the specified workflow descriptor and inputs, stage any WdlFiles into the workflow-inputs subdirectory
+   * of the workflow execution directory.  Return a Map of the input values with any input WdlFiles adjusted to
+   * reflect host paths.
+   *
+   * Original input path: /could/be/anywhere/input.bam
+   * Host inputs path: $PWD/cromwell-executions/some-workflow-name/0f00-ba4/workflow-inputs/input.bam
+   */
+  private def stageWorkflowInputs(descriptor: WorkflowDescriptor): HostInputs = {
+    val hostInputsPath = Paths.get(hostExecutionPath(descriptor).toFile.getAbsolutePath, "workflow-inputs")
+    descriptor.actualInputs.map { case (name, value) =>
+      val hostPathAdjustedValue = value match {
+        case WdlFile(originalPath) =>
+          val executionPath = Paths.get(hostInputsPath.toFile.getAbsolutePath, originalPath.getFileName.toString)
+          Files.copy(originalPath, executionPath)
```

Comment:
At some point we need to get creative and come up with a way not to copy files for local execution.  I can only imagine how frustrating that will be if you have gigabytes of input to operate on and the first thing you have to do is copy it all.